tests/PHPUnit/Integration/EmailValidatorTest.php

```diff
@@ -20,6 +23,40 @@ protected function isValid($email)
         return Piwik::isValidEmailString($email);
     }
 
+    private function getAllTlds()
+    {
+        /** @var Client $client */
+        $client = new Client("http://data.iana.org");
+        /** @var Request $request */
+        $request = $client->get('/TLD/tlds-alpha-by-domain.txt');
+        /** @var Response $response */
+        $response = $request->send();
+
+        $tlds = explode("\n", $response->getBody(true));
+        foreach ($tlds as $key => $tld) {
+            if (strpos($tld, '#') !== false || $tld == "") {
+                unset($tlds[$key]);
+            }
+        }
+        return $tlds;
+    }
+
+    public function test_allCurrentTlds(){
+        $tlds = $this->getAllTlds();
+        if (count($tlds) === 0) {
+            $this->markTestSkipped("Couldn't get TLD list");
+        }
+
+        foreach ($tlds as $key => $tld) {
+            if (strpos(mb_strtolower($tld), 'xn--') !== 0) {
+                $tld = mb_strtolower($tld);
+            }
+            $this->assertTrue(
+                $this->isValid('test@example.' . idn_to_utf8($tld))
+            );
```

Comment:
Can we also add some negative cases? I presume so far all the tests would make work even if the method only always returned true :) (apart from one other test)